ocaml/runtime/fiber.c

```diff
@@ -803,6 +833,17 @@ void caml_free_stack (struct stack_info* stack)
 
   CAMLassert(stack->magic == 42);
   CAMLassert(cache != NULL);
+
+#ifndef USE_MMAP_MAP_STACK
+#ifdef NATIVE_CODE
+  // CR xclerc for xclerc: check the value returned by mprotect?
+  int page_size = getpagesize();
+  mprotect((void *) ((char *) stack + page_size),
```

Comment:
Maybe we could add some macros for taking the `struct stack_info` and returning the lower and upper bounds of the guard page?  There are other occurrences where this would clean things up.